recipes-core/initrdscripts/files/ni_provisioning.safemode

```diff
@@ -317,8 +319,8 @@ install_grubenv()
 	grub-editenv $BOOTFS_MOUNTPOINT/grub/grubenv set "BIOSBootMode=efi"
 	grub-editenv $BOOTFS_MOUNTPOINT/grub/grubenv set "NITarget=$NI_TARGET"
 
-	# If target has a TPM chip, we need to enable an alternate boot flow
-	#  involving initramfs and bootflags dir
+	# If target has a TPM chip or runs on Hyper-V, we need to enable an alternate boot flow
+	# involving initramfs and bootflags dir
```

Comment:
What is the benefit of *not* enabling `initramfs`?

If we enable it on every target which has a TPM, it seems like we would hit this branch with relative frequency anyway. Should we just enable it on all targets?

Reply:
One benefit I can think of is slightly quicker boot time.
Also, I'm not sure if the initramfs is unloaded after booting rootfs but if its not then some RAM savings also.

Reply:
Oh and also RAM savings from not loading unnecessary drivers.

Reply:
OK. I suppose I don't have a hard argument for removing it; just looking for places to remove complexity.